docs/references/architecture/adr-119-dog-mempool-gossip.md

```diff
@@ -0,0 +1,287 @@
+- FLOOD - the current transaction gossiping protocol in CometBFT
+- CAT Mempool - the transaction gossiping proposed by Celestia
+- Limiting number of peers to send a transaction to - Experimental protocol in CometBFT
+- Etherium's transaction gossiping protocol.
+
+
+### FLOOD - the current transaction gossiping protocol in CometBFT
+
+Currently, CometBFT gossips each transaction to all the connected peers, except the sender
+of the transaction. Thus, for every transaction, CometBFT keeps a list of senders to make sure
+it is not sent again to a node that has sent it. 
+This does not prevent a node to send the transaction to a node that has sent it to the 
+sender, thus creating a lot of duplicates. 
+
+This is the exact problem DOG was designed to tackle.
+
+
+
+## Decision
+
+The CometBFT team has decided to implement the protocol on top of the existing mempool
+(`flood` ) dissemination protocol, but make it optional. The protocol can be activated/deactivated
+via a new config flag `enable_dog_protocol`. More details on this in the sections below. 
+
+## Detailed Design
+
+We start the design section with a general description of how the algorithm works. 
+
+CometBFT caches received transactions and stores the IDs of all the peers that have sent it in a `senders` list. When a transaction enters the mempool,
+if it was already received it can be retrieved from the cache. When this happens a certain number of times, the receiving node notifies the sender to stop forwarding any transaction from the peer 
+that had originally sent the transaction. The exact point when the notification is sent is defined below as a redundancy threshold. (TODO LiNK)
+
+Namely, let's assume that transaction `tx_1` was sent by node `B` to node `C` which forwarded it to `A`. `A` already has the transaction and notifies node `C` about this. Node `C` will look up the senders of transaction `tx_1` and in the future disable transaction forwarding from node `B` to node `A`.
+
+In reality, if node `C` ha received `tx_1` from multiple senders (`X`,`Y`, `B`), it will pick the first one that has sent it and disable that route. 
+
+Node `C` keeps a map of `disabledRoutes` per peer and uses it to determine whether a transaction should be forwarded. 
+
+Entries for a particular peer are reset if a peer sends a `Reset` message or if a peer disconnects. 
+
+### Redundancy control
+
+In an ideal setting, receiving a transaction more than once is not needed. But in a Byzantine setting, allowing for only one route for a transaction can introduce an attack vector. 
+
+Operators can thus configure a desired redundancy and the gossip protocol will adjust route disabling based on this setting. Redundancy is defined as the ratio between duplicate and unique transactions. 
+
+<!-- Operators can also define how much they tolerate the redundancy to deviate from the desired level. -->
+
+By default, the redundancy is set to `1` and impacts the gossiping of transactions as follows:
```

Comment:
Here I have more a question than anything.

I see a factor as something that starts from 1. There is not 0 factor. So when I read that the target redundancy is 1, I think that there is no redundancy.

We should understand which way to measure it is more clear to the users.

Notice that in practice this does not change anything, essentially, as now `redundancy := allReceivedTxs / firstTimeTx`, where `allReceivedTxs := firstTimeTx + duplicateTxs`.

Reply:
There's some confusion about the definition of redundancy level, which we don't define as a factor. Maybe this needs to fixed or clarified in the text.

The definition is `redundancy = duplicateTxs / firstTimeTxs`, and the spec defines special values for corner cases. That's why 0 is a valid value, even if we don't allow a target of 0 redundancy. The default target redundancy is 1.